awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx

```diff
@@ -1,14 +1,209 @@
+      await Promise.all(
+        selected.map(({ id }) => InstanceGroupsAPI.destroy(id))
+      );
+    }, [selected]),
+    {
+      qsConfig: QS_CONFIG,
+      allItemsSelected: isAllSelected,
+      fetchItems: fetchInstanceGroups,
+    }
+  );
+
+  const handleDelete = async () => {
+    await handleDeleteInstanceGroups();
+    setSelected([]);
+  };
+
+  const canAdd = actions && actions.POST;
+
+  function cannotDelete(item) {
+    return !item.summary_fields.user_capabilities.delete;
+  }
+
+  let itemsUnableToDelete = false;
+  itemsUnableToDelete = modifiedSelected
+    .filter(cannotDelete)
+    .filter(item => item.name !== 'tower')
+    .map(item => item.name)
+    .join(', ');
+
+  const pluralizedItemName = i18n._(t`Instance Groups`);
+
+  let errorMessageDelete = '';
+
+  if (itemsUnableToDelete) {
+    if (modifiedSelected.some(cannotDelete)) {
+      errorMessageDelete = i18n._(
+        t`You do not have permission to delete ${pluralizedItemName}: ${itemsUnableToDelete}. `
+      );
+    }
+  }
+
+  if (modifiedSelected.some(item => item.name === 'tower')) {
+    if (errorMessageDelete.length > 0) {
+      errorMessageDelete = errorMessageDelete.concat('\n');
+    }
+    errorMessageDelete = errorMessageDelete.concat(
+      i18n._(t`The tower instance group cannot be deleted.`)
+    );
+  }
+
   return (
-    <PageSection>
-      <Card>
-        <div>Instance Group List</div>
-      </Card>
-    </PageSection>
+    <>
+      <PageSection>
+        <Card>
+          <PaginatedDataList
+            contentError={contentError}
+            hasContentLoading={isLoading || deleteLoading}
+            items={instanceGroups}
+            itemCount={instanceGroupsCount}
+            pluralizedItemName={pluralizedItemName}
+            qsConfig={QS_CONFIG}
+            onRowClick={handleSelect}
+            renderToolbar={props => (
+              <DatalistToolbar
+                {...props}
+                showSelectAll
+                showExpandCollapse
+                isAllSelected={isAllSelected}
+                onSelectAll={isSelected =>
+                  setSelected(isSelected ? [...instanceGroups] : [])
+                }
+                qsConfig={QS_CONFIG}
+                additionalControls={[
+                  ...(canAdd
+                    ? [
+                        <ToolbarAddButton
+                          key="add"
+                          linkTo={`${match.url}/add`}
+                        />,
+                      ]
+                    : []),
+                  <ToolbarDeleteButton
+                    key="delete"
+                    onDelete={handleDelete}
+                    itemsToDelete={modifiedSelected}
+                    pluralizedItemName={i18n._(t`Instance Groups`)}
+                    errorMessage={
+                      modifiedSelected.some(item => item.name === 'tower')
+                        ? errorMessageDelete
+                        : ''
+                    }
```

Comment:
It looks like the prop `errorMessage` only returns `errorMessageDelete` when the instance group name is "tower". On lines ~115-130, the `if` statements perform many checks to build up the `errorMessageDelete` string, but since `errorMessage` only cares about one condition (name = "tower"), could we simplify how we build the error message?

Reply:
@marshmalien, we need to cover 3 different use cases. 

- Only `tower` instance selected
- `tower` instance and another `foo` instance, 
-  just `foo` instance.

I treated the last case as a regular case that we already have in our code basis. 

I will try to refactor and simplify as per your suggestion.

Reply:
@marshmalien, I refactored this logic. If the `tower` instance group is not selected the `errorMessageDelete` will be empty, and `ToolbarDeleteButton` will handle the error message as we currently do for all other cases. If the `tower` instance is selected, then logic to create the error message will evaluate inside `InstanceGroupList`.